pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -70,3 +84,106 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 
 	return r.createOrUpdateStatefulset(ctx)
 }
+
+func (r *reconciler) reconcileTLSSecret(ctx context.Context) error {
```

Comment:
IMO you could simplify the flow:

the delete is safe for not found, see  https://github.com/Dynatrace/dynatrace-operator/blob/4407c4ee51002a764ba82dd00eea46b635d09525/pkg/util/kubeobjects/internal/query/query.go#L74

so what you could do is the following:

```suggestion
func (r *reconciler) reconcileTLSSecret(ctx context.Context) error {
	query := k8ssecret.Query(r.client, r.client, log)

	if r.dk.Spec.Templates.ExtensionExecutionController.TlsRefName != ""  {  // you could also create a property for this
		err := query.Delete()
		if err != nil {
			return err
		}
		return nil
	}
	... // do the rest
	
```

Reply:
Fixed, you're right thanks!!